File: frontend/types/album.ts
export interface AlbumDto {
  id?: string;
  title: string;
  artist: string;
  coverArtUrl: string;
  releaseDate?: string;
  genre?: string;
  trackCount?: number;
  providerUrl?: string;
  itunesId?: string; // Sometimes provided directly, but we map it if needed
}

export interface AlbumSearchResponse {
  albums: AlbumDto[];
  total: number;
  limit: number;
  offset: number;
}

export interface AlbumCreateRequest {
  itunesId?: string;
  title: string;
  artist: string;
  coverArtUrl: string;
  releaseDate?: string;
  genre?: string;
  trackCount?: number;
}

export interface SavedAlbumResponse {
  id: string;
  itunesId?: string;
  title: string;
  artist: string;
  coverArtUrl: string;
  releaseDate?: string;
  genre?: string;
  trackCount?: number;
  personalRating?: number;
  notes?: string;
  createdAt: string;
}

File: backend/src/main/java/com/musiciq/backend/entity/Album.java
package com.musiciq.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(
        name = "albums",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"user_id", "itunes_id"})
        }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Album {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "itunes_id")
    private String itunesId;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String artist;

    @Column(name = "cover_art_url")
    private String coverArtUrl;

    @Column(name = "release_date")
    private String releaseDate;

    private String genre;

    @Column(name = "track_count")
    private Integer trackCount;

    @Column(name = "personal_notes", length = 1000)
    private String personalNotes;

    @Column(name = "personal_rating")
    private Integer personalRating;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;
}
